File: examples/android-compose/src/main/kotlin/OrderPage.kt
package com.example.orders.ui

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import androidx.lifecycle.viewmodel.compose.viewModel
import com.uitestprobe.compose.*

// MARK: Order Management Page

@Composable
fun OrderPage(viewModel: OrderViewModel = viewModel()) {
    val orders by viewModel.orders.collectAsState()
    val tableState by viewModel.tableState.collectAsState()
    var showCreateDialog by remember { mutableStateOf(false) }

    Column(
        modifier = Modifier
            .fillMaxSize()
            .probeId("order-page")
            .probeType(ProbeType.PAGE)
    ) {
        // -- Toolbar --
        Row(
            modifier = Modifier.fillMaxWidth().padding(16.dp),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically
        ) {
            Text("Order Management", style = MaterialTheme.typography.headlineSmall)

            Button(
                onClick = { showCreateDialog = true },
                modifier = Modifier
                    .probeId("create-order-btn")
                    .probeType(ProbeType.ACTION)
                    .probeTrigger(TriggerType.CLICK, opens = "create-order-modal")
            ) {
                Text("Create Order")
            }
        }

        // -- Status Filter --
        StatusFilterDropdown(
            selected = viewModel.statusFilter,
            onSelect = { viewModel.setFilter(it) },
            modifier = Modifier
                .padding(horizontal = 16.dp)
                .probeId("status-filter")
                .probeType(ProbeType.SELECTOR)
                .probeLinkage(
                    target = "order-table",
                    effect = LinkageEffect.DATA_RELOAD,
                    path = ApiPath("GET /api/orders")
                )
        )

        // -- Order Table --
        LazyColumn(
            modifier = Modifier
                .weight(1f)
                .probeId("order-table")
                .probeType(ProbeType.DATA_CONTAINER)
                .probeSource("GET /api/orders")
                .probeState(tableState)
                .probeChildren("order-row-*")
        ) {
            items(orders, key = { it.id }) { order ->
                OrderRow(
                    order = order,
                    modifier = Modifier
                        .probeId("order-row-${order.id}")
                        .probeType(ProbeType.DATA_ROW)
                )
            }
        }

        // -- Paginator --
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(16.dp)
                .probeId("paginator")
                .probeType(ProbeType.NAVIGATION)
                .probeLinkage(
                    target = "order-table",
                    effect = LinkageEffect.DATA_RELOAD,
                    path = ApiPath("GET /api/orders")
                ),
            horizontalArrangement = Arrangement.Center,
            verticalAlignment = Alignment.CenterVertically
        ) {
            TextButton(
                onClick = { viewModel.prevPage() },
                enabled = viewModel.page > 1
            ) { Text("Previous") }

            Text("Page ${viewModel.page} of ${viewModel.totalPages}")

            TextButton(
                onClick = { viewModel.nextPage() },
                enabled = viewModel.page < viewModel.totalPages
            ) { Text("Next") }
        }
    }

    // -- Create Order Modal --
    if (showCreateDialog) {
        CreateOrderDialog(
            onDismiss = { showCreateDialog = false },
            onSubmit = { draft ->
                viewModel.submitOrder(draft)
                showCreateDialog = false
            },
            modifier = Modifier
                .probeId("create-order-modal")
                .probeType(ProbeType.MODAL)
        )
    }
}

@Composable
private fun CreateOrderDialog(
    onDismiss: () -> Unit,
    onSubmit: (OrderDraft) -> Unit,
    modifier: Modifier = Modifier
) {
    var customer by remember { mutableStateOf("") }
    var amount by remember { mutableStateOf("") }

    AlertDialog(
        onDismissRequest = onDismiss,
        modifier = modifier,
        title = { Text("New Order") },
        text = {
            Column(
                modifier = Modifier
                    .probeId("create-order-form")
                    .probeType(ProbeType.FORM)
            ) {
                OutlinedTextField(
                    value = customer,
                    onValueChange = { customer = it },
                    label = { Text("Customer") },
                    modifier = Modifier
                        .fillMaxWidth()
                        .probeId("input-customer")
                        .probeType(ProbeType.FORM_FIELD)
                        .probeValidation(ValidationType.REQUIRED, "Customer is required")
                )
                Spacer(Modifier.height(8.dp))
                OutlinedTextField(
                    value = amount,
                    onValueChange = { amount = it },
                    label = { Text("Amount") },
                    modifier = Modifier
                        .fillMaxWidth()
                        .probeId("input-amount")
                        .probeType(ProbeType.FORM_FIELD)
                        .probeValidation(ValidationType.RANGE_MIN, "Must be > 0", min = 0.01)
                )
            }
        },
        confirmButton = {
            Button(
                onClick = { onSubmit(OrderDraft(customer, amount.toDoubleOrNull() ?: 0.0)) },
                modifier = Modifier
                    .probeId("submit-order-btn")
                    .probeType(ProbeType.ACTION)
                    .probeLinkage(
                        target = "order-table",
                        effect = LinkageEffect.DATA_RELOAD,
                        path = ApiPath("POST /api/orders")
                    )
            ) { Text("Submit") }
        },
        dismissButton = {
            TextButton(onClick = onDismiss) { Text("Cancel") }
        }
    )
}

File: examples/ios-swiftui/Sources/OrderPage.swift
import SwiftUI
import UITestProbe

// MARK: - Order Management Page

struct OrderPage: View {
    @StateObject private var viewModel = OrderViewModel()

    var body: some View {
        VStack(spacing: 0) {
            // -- Toolbar --
            HStack {
                Text("Order Management")
                    .font(.title2.bold())

                Spacer()

                Button("Create Order") {
                    viewModel.showCreateDialog = true
                }
                .probeId("create-order-btn")
                .probeType(.action)
                .probeTrigger(.tap, opens: "create-order-modal")
            }
            .padding()

            // -- Status Filter --
            Picker("Status", selection: $viewModel.statusFilter) {
                Text("All").tag(OrderStatus?.none)
                ForEach(OrderStatus.allCases) { status in
                    Text(status.label).tag(Optional(status))
                }
            }
            .probeId("status-filter")
            .probeType(.selector)
            .probeLinkage(
                to: "order-table",
                effect: .dataReload,
                path: .api("GET /api/orders")
            )
            .padding(.horizontal)

            // -- Order Table --
            List(viewModel.orders) { order in
                OrderRow(order: order)
            }
            .probeId("order-table")
            .probeType(.dataContainer)
            .probeSource("GET /api/orders")
            .probeState(viewModel.tableState)
            .probeChildren(["order-row-*"])

            // -- Paginator --
            HStack {
                Button("Previous") { viewModel.prevPage() }
                    .disabled(viewModel.page <= 1)

                Text("Page \(viewModel.page) of \(viewModel.totalPages)")

                Button("Next") { viewModel.nextPage() }
                    .disabled(viewModel.page >= viewModel.totalPages)
            }
            .probeId("paginator")
            .probeType(.navigation)
            .probeLinkage(
                to: "order-table",
                effect: .dataReload,
                path: .api("GET /api/orders")
            )
            .padding()
        }
        .sheet(isPresented: $viewModel.showCreateDialog) {
            CreateOrderModal(viewModel: viewModel)
                .probeId("create-order-modal")
                .probeType(.modal)
        }
        .probeId("order-page")
        .probeType(.page)
        .onAppear { viewModel.loadOrders() }
    }
}

// MARK: - Order Row

struct OrderRow: View {
    let order: Order

    var body: some View {
        HStack {
            Text(order.id).fontWeight(.medium)
            Spacer()
            Text(order.customer)
            Spacer()
            Text(order.status.label)
                .foregroundColor(order.status.color)
            Spacer()
            Text(order.total, format: .currency(code: "USD"))
        }
        .probeId("order-row-\(order.id)")
        .probeType(.dataRow)
    }
}

// MARK: - Create Order Modal

struct CreateOrderModal: View {
    @ObservedObject var viewModel: OrderViewModel
    @Environment(\.dismiss) private var dismiss

    var body: some View {
        NavigationStack {
            Form {
                TextField("Customer", text: $viewModel.draft.customer)
                    .probeId("input-customer")
                    .probeType(.formField)
                    .probeValidation(.required, message: "Customer is required")

                TextField("Amount", value: $viewModel.draft.total, format: .number)
                    .probeId("input-amount")
                    .probeType(.formField)
                    .probeValidation(.range(min: 0.01), message: "Must be > 0")
            }
            .probeId("create-order-form")
            .probeType(.form)
            .toolbar {
                ToolbarItem(placement: .confirmationAction) {
                    Button("Submit") { viewModel.submitOrder(); dismiss() }
                        .probeId("submit-order-btn")
                        .probeType(.action)
                        .probeLinkage(
                            to: "order-table",
                            effect: .dataReload,
                            path: .api("POST /api/orders")
                        )
                }
                ToolbarItem(placement: .cancellationAction) {
                    Button("Cancel") { dismiss() }
                }
            }
            .navigationTitle("New Order")
        }
    }
}
